### PR TITLE
(refactor) Update PatientProgram interface to display program workflows and states

### PR DESCRIPTION
**Pull request by CynthiaKamau**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [ ] My work includes tests or is validated by existing tests.

## Summary

This PR updates the [PatientProgram](https://github.com/openmrs/openmrs-esm-patient-chart/blob/main/packages/esm-patient-common-lib/src/types/index.ts#L21) type annotation to accommodate program workflows and states.

### CODE REVIEW

File: packages/esm-patient-common-lib/src/types/index.ts [hunk 1]
```diff
@@ -20,10 +20,45 @@ export interface DashboardConfig extends DashboardLinkConfig {
...
+  outcome?: null;
+  states?: [];
+  links: Links;
+  resourceVersion?: string;
+}
+
+export type Links = Array<{
+  rel: string;
+  uri: string;
+}>;
```
Comment (Vijaykv5) [inline packages/esm-patient-common-lib/src/types/index.ts:58]: do we have a field `resourceAlias` for Links?

Comment (Vijaykv5) [inline packages/esm-patient-common-lib/src/types/index.ts:58]: @CynthiaKamau do we have a field `resourceAlias` for Links?

Comment (CynthiaKamau) [reply packages/esm-patient-common-lib/src/types/index.ts:58]: Yes, please refer to the rest endpoint response [here](https://rest.openmrs.org/#list-program-enrollment-by-uuid)

File: packages/esm-patient-common-lib/src/types/index.ts [hunk 2]
```diff
@@ -20,10 +20,45 @@ export interface DashboardConfig extends DashboardLinkConfig {
...
-  program: OpenmrsResource;
   dateEnrolled: string;
-  dateCompleted: string;
-  location: OpenmrsResource;
+  dateCompleted: string | null;
+  location?: {
+    uuid: string;
+    display: string;
+    links: Links;
+  };
```
Comment (denniskigen) [inline packages/esm-patient-common-lib/src/types/index.ts:47]:
```suggestion
  location?: DisplayMetadata;
```

**Review by denniskigen — APPROVED** [review]
LGTM. Thanks.

File: packages/esm-patient-common-lib/src/types/index.ts [hunk 3]
```diff
@@ -20,10 +20,45 @@ export interface DashboardConfig extends DashboardLinkConfig {
...
   uuid: string;
+  patient?: DisplayMetadata;
+  program: {
+    uuid: string;
+    name: string;
+    allWorkflows: Array<{
+      uuid: string;
+      concept: DisplayMetadata;
+      retired: boolean;
+      states: Array<{}>;
```
Comment (denniskigen) [inline packages/esm-patient-common-lib/src/types/index.ts:31]:
```suggestion
      states: Array<Record<string, unknown>>;
```

Comment (denniskigen) [reply packages/esm-patient-common-lib/src/types/index.ts:31]: https://rest.openmrs.org/#list-programs

File: packages/esm-patient-common-lib/src/types/index.ts [hunk 4]
```diff
@@ -20,10 +20,45 @@ export interface DashboardConfig extends DashboardLinkConfig {
...
+export type Links = Array<{
+  rel: string;
+  uri: string;
+}>;
+
+export interface DisplayMetadata {
+  display?: string;
+  links?: Links;
+  uuid?: string;
 }
```
Comment (denniskigen) [inline packages/esm-patient-common-lib/src/types/index.ts:64]: Maybe these shouldn't be optional properties?

**Review by lucyjemutai — APPROVED** [review]

File: packages/esm-patient-common-lib/src/types/index.ts (hunk 4 above)
Comment (CynthiaKamau) [reply packages/esm-patient-common-lib/src/types/index.ts:64]: Done

**Review by arodidev — APPROVED** [review]
Thanks @CynthiaKamau LGTM

**Pull request merged**